site/layouts/shortcodes/table.html

```diff
@@ -20,10 +24,17 @@
   {{- $table_content = partialCached "table-content" . -}}
 {{- end -}}
 
-{{- $table := printf "<table%s>\n%s</table>" $table_attributes $table_content -}}
+{{- with $caption -}}
+  {{- $table_caption = printf `%s` . -}}
+{{- end -}}
+
+{{- $table_caption_attributes = printf ` class="visually-hidden"` -}}
+
+{{- $table := printf "\n<table%s>\n  <caption%s>%s</caption>\n%s</table>\n" $table_attributes $table_caption_attributes $table_caption $table_content -}}
 
 <div class="bd-example">
   <table{{ with $class }} class="{{ . }}"{{ end }}>
+    <caption class="visually-hidden">{{ with $table_caption }}{{.}}{{ end }}</caption>
```

Comment:
Isn't it better to remove entirely the `<caption>` if no content has been provided?

When it's called like this for example: `{{< table class="table" simplified="false" >}}`. Otherwise it's generated like this:

```
<caption class="visually-hidden">...</caption>
```